core/rawdb/freezer_table.go

```diff
@@ -227,6 +236,9 @@ func (t *freezerTable) repair() error {
 	}
 	// Ensure the index is a multiple of indexEntrySize bytes
 	if overflow := stat.Size() % indexEntrySize; overflow != 0 {
+		if t.readonly {
```

Comment:
Do we need this check? The index file is read only and truncation will fail anyway.

Reply:
Removed, also the other unnecessary checks.